src/pages/Download.js

```diff
@@ -33,299 +33,397 @@ const Download = () => {
           <div className={'flex flex-col items-center pt-4'}>
             <img src={logo_name} alt={'UPBGE Logo with name'} />
             <br />
-            <SectionTitle>Stable Release Version 0.3</SectionTitle>
-            <p>Released 4 December 2021</p>
+            <SectionTitle>Version Info</SectionTitle>
```

Comment:
This part I prefer to leave as currently: 
Stable Release Version 0.3
Released 4 December 2021

And below this we can add the block as you did. Maybe indicating 2 options only:
UPBGE 0.3+ is based on current Blender master
UPBGE 0.2.5b is based on legacy Blender 2.79.7